client/src/main/java/io/pravega/client/stream/impl/EventStreamReaderImpl.java

```diff
@@ -201,20 +213,19 @@ private String updateGroupStateIfNeeded() throws ReaderNotInReaderGroupException
                     releaseSegmentsIfNeeded(position);
                     atCheckpoint = null;
                 }
-                return null;
+                checkpoint = groupState.getCheckpoint();
```

Comment:
This change would cause to complete checkpointing of all the pending silent checkpoints for this reader. In the older code the application would have to invoke `readNextEvent()` again to clear this, right ?

Reply:
yes